### PR TITLE
updates to standardise execution env use pages - macos

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
Updating "Using ..." pages for all execution environments to standardise how this information is formatted and presented. This is the macOS PR, there will be a PR for each execution environment.
 
In this PR I've moved a lot of "usage" detail out of various tutorial pages into the main using-macos guide. This might make the PR hard to follow.

# Reasons
https://circleci.atlassian.net/browse/DOCSTEAM-326?atlOrigin=eyJpIjoiYmYxYzVkNDRiZTlkNGQxOWJhYTIzMDIxMzgyNjJkODAiLCJwIjoiaiJ9

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: https://circleci.com/docs/style/.

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [x] Break up walls of text by adding paragraph breaks.
- [x] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [x] Keep the title between 20 and 70 characters.
- [x] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [x] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [x] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [x] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/using-macos.md [hunk 1]
```diff
@@ -4,38 +4,51 @@ title: "Using the macOS execution environment"
...
+      - run: xcodebuild -version
+```
+
+## Supported Xcode versions
+{: #supported-xcode-versions }
+
+{% include snippets/xcode-versions.md %}
+
+## Available resource classes
+{: available-resource-classes }
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-macos.md:31]:
```suggestion
{: #available-resource-classes }
```

File: jekyll/_cci2/using-macos.md [hunk 2]
```diff
@@ -112,6 +137,204 @@ straightforward. To build ARM64 binaries, prepend the `xcodebuild` command with
...
+
+### React Native projects
+{: #react-native-projects }
+
+React Native projects can be built on CircleCI using `macos` and `docker` executor types. For an example of configuring a React Native project, please see [our demo React Native application](https://github.com/CircleCI-Public/circleci-demo-react-native)
+
+### Creating a `config.yml` File
+{: #creating-a-configyml-file }
+
+The most flexible way to customize your build is to modify the CircleCI configuration for your project in `.circleci/config.yml`. This allows you to run arbitrary bash commands as well as utilise built-in features such as workspaces and caching. See the [Configuring CircleCI]( {{ site.baseurl }}/2.0/configuration-reference/) documentation for a detailed description of the structure of the `config.yml` file.
```
Comment (devinmarieb) [inline jekyll/_cci2/using-macos.md:272]:
```suggestion
The most flexible way to customize your build is to modify the CircleCI configuration for your project in `.circleci/config.yml`. This allows you to run arbitrary bash commands as well as use built-in features such as workspaces and caching. See the [Configuring CircleCI]( {{ site.baseurl }}/2.0/configuration-reference/) documentation for a detailed description of the structure of the `.circleci/config.yml` file.
```

File: jekyll/_cci2/using-macos.md [hunk 3]
```diff
@@ -47,11 +60,54 @@ Xcode Version | Class                 | Minimum Available Storage
...
 11.*          | macos.x86.medium.gen2 | 23GB
 12.*          | medium, large         | 30GB
-12.*          | macos.x86.medium.gen2 | 30GB<sup>(2)</sup>
+12.*          | macos.x86.medium.gen2 | 30GB
 13.*          | medium, large         | 23GB
 13.*          | macos.x86.medium.gen2 | 89GB
 {: class="table table-striped"}
 
-<sup>(1)</sup> _This resource requires a minimum 24-hour lease. See the [Dedicated Host for macOS]({{ site.baseurl }}/2.0/dedicated-hosts-macos) page to learn more about this resource class._
+**Note:** Xcode 12.0.1, 12.4.0 and 12.5.1 have a minimum 100GB of available storage.
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/using-macos.md:68]: Good spot this is confusing! I will find out and clear this up

[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/using-macos.md:68]: This table is showing the available storage on the VM once all the macOS gubbins is installed. The note is calling out the exceptions that for these Xcode versions there will be at least 100GB of storage available. I'll update the text, let me know if it's still unclear.

File: jekyll/_cci2/using-macos.md [hunk 4]
```diff
@@ -4,38 +4,51 @@ title: "Using the macOS execution environment"
...
-      # with Xcode 12.5.1 installed
-      - run: xcodebuild -version
+    resource_class: large
 ```
 
-### macOS VM Storage
+## macOS VM Storage
 {: #macos-vm-storage }
 
 The amount of available storage on our macOS VMs depends on the resource class and Xcode image being used. The size of the Xcode images varies based on which tools are pre-installed.
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-macos.md:54]:
```suggestion
The amount of available storage on CircleCI macOS virtual machines depends on the resource class and Xcode image being used. The size of the Xcode images varies based on which tools are pre-installed. The table below indicated how much storage will be available with various Xcode/resource class combinations. Also note the exceptions to this noted below.
```

File: jekyll/_cci2/using-macos.md [hunk 5]
```diff
@@ -47,11 +60,282 @@ Xcode Version | Class                 | Minimum Available Storage
...
 11.*          | macos.x86.medium.gen2 | 23GB
 12.*          | medium, large         | 30GB
-12.*          | macos.x86.medium.gen2 | 30GB<sup>(2)</sup>
+12.*          | macos.x86.medium.gen2 | 30GB
 13.*          | medium, large         | 23GB
 13.*          | macos.x86.medium.gen2 | 89GB
 {: class="table table-striped"}
 
-<sup>(1)</sup> _This resource requires a minimum 24-hour lease. See the [Dedicated Host for macOS]({{ site.baseurl }}/2.0/dedicated-hosts-macos) page to learn more about this resource class._
+**Note:** Xcode 12.0.1, 12.4.0 and 12.5.1 have a minimum 100GB of available storage.
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-macos.md:68]:
```suggestion
**Note:** If you specify Xcode 12.0.1, 12.4.0 and 12.5.1 you have a minimum 100GB of available storage.
```

File: jekyll/_cci2/using-macos.md [hunk 6]
```diff
@@ -67,6 +67,31 @@ Xcode Version | Class                 | Minimum Available Storage
...
+
+To read about our customer support policy regarding beta images, please check out the following [support center article](https://support.circleci.com/hc/en-us/articles/360046930351-What-is-CircleCI-s-Xcode-Beta-Image-Support-Policy-).
+
+### Apple silicon support
+{: #apple-silicon-support }
+
+It is possible to build Apple Silicon/Universal binaries using Xcode `12.0.0` and higher as Apple provides both the Intel (`x86_64`) and Apple Silicon (`arm64`) toolchains in this release. Cross-compiling Apple Silicon binaries on Intel hosts has an additional overhead and as a result compilation times will be longer than native compilation for Intel.
+
+Running or testing Apple Silicon apps natively is currently not possible as CircleCI build hosts are Intel-based Macs. Binaries will need to be exported as [artifacts]({{site.baseurl}}/2.0/artifacts/) for testing apps locally. Alternatively, [CircleCI runner]({{site.baseurl}}/2.0/runner-overview/#supported) can also be used to run jobs natively on Apple Silicon.
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/using-macos.md:87]: I have not checked but I will!

[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/using-macos.md:87]: Yes still correct, checked with the execution team

File: jekyll/_cci2/using-macos.md [hunk 7]
```diff
@@ -67,6 +67,31 @@ Xcode Version | Class                 | Minimum Available Storage
 
 **Note:** Xcode 12.0.1, 12.4.0 and 12.5.1 have a minimum 100GB of available storage.
 
+## Image update cycle for the macOS executor
+{: #using-the-macos-executor }
+
+Each `macos` job is run a fresh virtual machine, running a specified version macOS. We build a new image each time a new stable, or beta, version of Xcode is released by Apple and aim to get these deployed as soon as possible. Generally, the contents of a particular build image will remain unchanged, except in very exceptional circumstances we might be forced to re-build a container for a specific reason. Our goal is to keep your execution environment stable, and to allow you to opt-in to newer containers by setting the `xcode` key in your `config.yml` file.
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/using-macos.md:65]: Thanks for pointing this out!

File: jekyll/_cci2/using-macos.md [hunk 8]
```diff
@@ -35,10 +22,28 @@ jobs:
...
+    macos:
+      xcode: 12.5.1
+    resource_class: large
+```
+
+## macOS VM Storage
 {: #macos-vm-storage }
 
-The amount of available storage on our macOS VMs depends on the resource class and Xcode image being used. The size of the Xcode images varies based on which tools are pre-installed.
+The amount of available storage on CircleCI macOS virtual machines depends on the resource class and Xcode image being used. The size of the Xcode images varies based on which tools are pre-installed. The table below indicated how much storage will be available with various Xcode/resource class combinations. Also note the exceptions to this noted below.
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-macos.md:46]:
```suggestion
The amount of available storage on CircleCI macOS virtual machines depends on the resource class and Xcode image being used. The size of the Xcode images varies based on which tools are pre-installed. The table below indicates how much storage will be available with various Xcode/resource class combinations. Also note the exceptions to this noted below.
```

File: jekyll/_cci2/using-macos.md [hunk 9]
```diff
@@ -35,10 +22,28 @@ jobs:
...
+    macos:
+      xcode: 12.5.1
+    resource_class: large
+```
+
+## macOS VM Storage
 {: #macos-vm-storage }
 
-The amount of available storage on our macOS VMs depends on the resource class and Xcode image being used. The size of the Xcode images varies based on which tools are pre-installed.
+The amount of available storage on CircleCI macOS virtual machines depends on the resource class and Xcode image being used. The size of the Xcode images varies based on which tools are pre-installed. The table below indicates how much storage will be available with various Xcode/resource class combinations. Also note the exceptions to this noted below.
```
Comment (rosieyohannan) [inline jekyll/_cci2/using-macos.md:46]: FYI I made some changes to this intro to the table, and to the phrasing of the note to try and clear up the exceptions
